internal/controller/datadogagent/feature/admissioncontroller/feature.go

```diff
@@ -99,11 +99,11 @@ func (f *admissionControllerFeature) Configure(dda metav1.Object, ddaSpec *v2alp
 	ac := ddaSpec.Features.AdmissionController
 
 	if ac != nil && apiutils.BoolValue(ac.Enabled) {
-		if ac.Validation != nil && ac.Validation.Enabled != nil {
```

Comment:
I think the intention here is not to override cluster-agent defaults when the setting in the operator is not configured.

So something more like:

```go
if ac.Validation != nil {
  f.validationWebhookConfig = &ValidationConfig{
    enabled: apiutils.BoolValue(ac.Validation.Enabled),
  }
}
```